Assets/_Scripts/PlayerPickUpItem.cs

```diff
@@ -12,17 +12,15 @@ public class PlayerPickUpItem : MonoBehaviour {
 		pd = GetComponent<PlayerData> ();
 	}
 	// Update is called once per frame
-	public void PickUp (int mode, bool isBuff, SpellDB.AttackID ID) {
-		if (isBuff)
-			BH.UpdateBuff (mode);
+	public void PickUp (bool buff, bool TriggerNow, SpellDB.AttackID ID) {
```

Comment:
My capture is this script barely do anything.
I would consider merge this script with the PlayerBuffStatus. I don't see the reason you need this layer